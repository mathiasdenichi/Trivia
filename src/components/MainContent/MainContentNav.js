import React from 'react'
import PropTypes from 'prop-types'

const Nav = ({ reset, checkAnswer }) => (
  <div id='navcontainer'>
    <div id='nav'>
      <button
        onClick={() => checkAnswer('True')}
        id='trueBtn'
      >
      True
      </button>
      <button
        onClick={() => checkAnswer('False')}
        id='falseBtn'
      >
    False
      </button>
    </div>
    <button
      onClick={reset}
      id='resetBtn'
    >
      Reset
    </button>
  </div>
)

Nav.propTypes = {
  reset: PropTypes.func,
  checkAnswer: PropTypes.func,
}

Nav.defaultProps = {
  reset: (value) => { this.setState({ value }) },
  checkAnswer: (value) => { value.find() },
}

export default Nav
