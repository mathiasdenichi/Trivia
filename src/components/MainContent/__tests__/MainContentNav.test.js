import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MainContentNav from '../MainContentNav'


configure({ adapter: new Adapter() })

describe('Main Content Nav', () => {
  const wrapper = shallow(<MainContentNav />)
  it('renders the component and matches it to snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
