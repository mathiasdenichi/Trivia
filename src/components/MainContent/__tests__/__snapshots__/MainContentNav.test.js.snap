// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Content Nav renders the component and matches it to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    checkAnswer={[Function]}
    reset={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="nav"
        >
          <button
            id="trueBtn"
            onClick={[Function]}
          >
            True
          </button>
          <button
            id="falseBtn"
            onClick={[Function]}
          >
            False
          </button>
        </div>,
        <button
          id="resetBtn"
          onClick={[Function]}
        >
          Reset
        </button>,
      ],
      "id": "navcontainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              id="trueBtn"
              onClick={[Function]}
            >
              True
            </button>,
            <button
              id="falseBtn"
              onClick={[Function]}
            >
              False
            </button>,
          ],
          "id": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "True",
              "id": "trueBtn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "True",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "False",
              "id": "falseBtn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "False",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset",
          "id": "resetBtn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Reset",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="nav"
          >
            <button
              id="trueBtn"
              onClick={[Function]}
            >
              True
            </button>
            <button
              id="falseBtn"
              onClick={[Function]}
            >
              False
            </button>
          </div>,
          <button
            id="resetBtn"
            onClick={[Function]}
          >
            Reset
          </button>,
        ],
        "id": "navcontainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                id="trueBtn"
                onClick={[Function]}
              >
                True
              </button>,
              <button
                id="falseBtn"
                onClick={[Function]}
              >
                False
              </button>,
            ],
            "id": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "True",
                "id": "trueBtn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "True",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "False",
                "id": "falseBtn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "False",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset",
            "id": "resetBtn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reset",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
